config/v2/config.go

```diff
@@ -260,6 +260,8 @@ type ComponentModule struct {
 	ProvidersMap map[string]string `yaml:"providers,omitempty"`
 	// For Each metadata argument https://developer.hashicorp.com/terraform/language/modules/syntax#meta-arguments
 	ForEach *string `yaml:"for_each,omitempty"`
+	//
+	DependsOn []string `yaml:"depends_on,omitempty"`
```

Comment:
not sure if it should also be added to the legacy case where a component uses a single moduleSource?

https://github.com/vincenthsh/fogg/blob/37611aef7adb6bfbf584a4db997c25c0002b84f7/config/v2/config.go#L235-L237